File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int minutesInOven) => ExpectedMinutesInOven() - minutesInOven;

    public int PreparationTimeInMinutes(int numberOfLayers) => numberOfLayers * 2;

    public int ElapsedTimeInMinutes(int numberOfLayers, int minutesInOven) => PreparationTimeInMinutes(numberOfLayers) + minutesInOven;
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven actualMinutes = expectedMinutesInOven - actualMinutes

let preparationTimeInMinutes numberOfLayers = numberOfLayers * 2

let elapsedTimeInMinutes numberOfLayers actualMinutes = preparationTimeInMinutes numberOfLayers + actualMinutes
